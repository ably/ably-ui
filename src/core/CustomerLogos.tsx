import React from "react";

type CompanyEntity = {
  label: string;
  logo: string;
};

type CustomerLogosProps = {
  companies: CompanyEntity[];
  additionalCss?: string;
};

const CustomerLogos = ({
  companies,
  additionalCss = "",
}: CustomerLogosProps) => {
  return (
    <section className="w-full bg-white">
      <ul
        className={`py-16 flex flex-row flex-wrap md:flex-nowrap content-between m-auto items-center ${additionalCss}`}
      >
        {companies.map((company) => (
          <li
            key={company.label}
            className="flex-auto text-center sm:w-1/3 w-1/2"
          >
            <img alt={company.label} src={company.logo} className="mx-auto" />
          </li>
        ))}
      </ul>
    </section>
  );
};

export default CustomerLogos;
