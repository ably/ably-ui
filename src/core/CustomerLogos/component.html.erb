<section class="w-full bg-white">
  <%= tag.ul class: ["py-64 flex flex-row flex-wrap md:flex-nowrap content-between m-auto items-center", @additional_css].join(' ') do %>
    <% companies.each do |company| %>
	    <li class="flex-auto text-center sm:w-1/3 w-1/2">
	      <%= image_tag company[:logo], alt: "#{company[:label]} logo", class: 'mx-auto' %>
	    </li>
    <% end %>
  <% end %>
</section>
