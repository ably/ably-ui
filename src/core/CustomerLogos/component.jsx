import React from "react";
import T from "prop-types";

const CustomerLogos = ({ companies }) => {
  return (
    <section className="w-full bg-white">
      <ul className="py-64 flex flex-row flex-wrap md:flex-nowrap content-between m-auto items-center">
        {companies.map((company) => (
          <li key={company.label} className="flex-auto text-center sm:w-1/3 w-1/2">
            <img alt={company.label} src={company.logo} className="mx-auto" />
          </li>
        ))}
      </ul>
    </section>
  );
};

CustomerLogos.propTypes = {
  companies: T.arrayOf(
    T.shape({
      label: T.string,
      logo: T.string,
    })
  ),
};

export default CustomerLogos;
