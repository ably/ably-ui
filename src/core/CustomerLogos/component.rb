require 'json'

module AblyUi
  module Core
    class CustomerLogos < ViewComponent::Base
      attr_reader :companies

      def initialize(companies:, additional_css: "")
        @companies = companies
        @additional_css = additional_css
      end
    end
  end
end
