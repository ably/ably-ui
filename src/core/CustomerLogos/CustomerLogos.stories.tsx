import CustomerLogos from "../CustomerLogos";

import hubspot from "../images/cust-logo-hubspot-mono-pos.svg";
import webflow from "../images/cust-logo-webflow-col-pos.svg";
import mentimeter from "../images/cust-logo-mentimeter-mono-pos.svg";
import toyota from "../images/cust-logo-toyota-mono-pos.svg";
import split from "../images/cust-logo-split-mono-pos.svg";
import australian from "../images/cust-logo-ausopen-mono-pos.svg";

export default {
  title: "JS Components/Customer Logos",
  component: CustomerLogos,
  args: {
    companies: [
      {
        label: "Hubspot",
        logo: hubspot,
      },
      {
        label: "Webflow",
        logo: webflow,
      },
      {
        label: "Mentimeter",
        logo: mentimeter,
      },
      {
        label: "Toyota",
        logo: toyota,
      },
      {
        label: "Split",
        logo: split,
      },
      {
        label: "Australian Open",
        logo: australian,
      },
    ],
  },
};

export const Default = {};
