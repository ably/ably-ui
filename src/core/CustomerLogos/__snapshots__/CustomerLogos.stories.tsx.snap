// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JS Components/Customer Logos Default smoke-test 1`] = `
<section class="w-full bg-white">
  <ul class="py-64 flex flex-row flex-wrap md:flex-nowrap content-between m-auto items-center ">
    <li class="flex-auto text-center sm:w-1/3 w-1/2">
      <img alt="Hubspot"
           src="data:image/svg+xml,%3csvg%20width='244'%20height='122'%20viewBox='0%200%20244%20122'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M67.1612%2058.7658V47.5718H72.042V74.2929H67.1612V63.4624H56.8808V74.2929H52V47.5718H56.8808V58.7658H67.1612ZM83.6246%2069.6931C85.8546%2069.6931%2087.6624%2067.8853%2087.6624%2065.6553V54.2102H92.2753V65.6553C92.2753%2070.4376%2088.3985%2074.3144%2083.6162%2074.3144C78.8339%2074.3144%2074.9571%2070.4376%2074.9571%2065.6553V54.2102H79.5867V65.6553C79.5867%2067.8853%2081.3945%2069.6931%2083.6246%2069.6931ZM121.964%2055.3891C121.964%2053.0432%20123.516%2052.2995%20125.216%2052.2995C126.587%2052.2995%20128.397%2053.3421%20129.581%2054.6072L132.613%2051.0345C131.099%2048.9875%20128.029%2047.5718%20125.515%2047.5718C120.493%2047.5718%20116.861%2050.5132%20116.861%2055.3891C116.861%2060.5643%20120.48%2061.8399%20123.577%2062.9316C125.894%2063.7483%20127.919%2064.4621%20127.919%2066.6286C127.919%2068.1901%20126.403%2069.5699%20124.666%2069.5699C122.179%2069.5699%20121.219%2068.465%20120.121%2067.2015C120.011%2067.074%20119.899%2066.9448%20119.783%2066.8151L116.419%2070.3136C118.571%2072.9561%20121.225%2074.2977%20124.406%2074.2977C129.176%2074.2977%20133.015%2071.3204%20133.015%2066.6668C133.015%2060.6108%20128.994%2059.3409%20125.802%2058.3326C123.7%2057.6687%20121.957%2057.1181%20121.957%2055.3891H121.964ZM186.859%2067.0279C186.859%2068.8406%20187.636%2070.0243%20190.371%2070.0243V70.0195C190.89%2070.0195%20191.483%2069.9836%20192%2069.9478L190.855%2074.1637C190.244%2074.2736%20189.624%2074.3233%20189.004%2074.312C185.048%2074.312%20182.164%2072.2052%20182.164%2068.0251V51.6825L186.859%2049.5757V54.9396H191.11V59.0049H186.859V67.0279ZM99.844%2056.5274C101.391%2055.0089%20102.991%2054.3441%20105.284%2054.3441V54.3465C110.462%2054.3465%20114.914%2058.6749%20114.914%2064.3328C114.914%2070.0626%20110.077%2074.3168%20104.576%2074.3168C99.6192%2074.3168%2095.2071%2070.1367%2095.2071%2064.0339V47.732H99.844V56.5274ZM99.9946%2064.3902C99.9946%2067.2958%20102.35%2069.6512%20105.256%2069.6512C108.161%2069.6512%20110.517%2067.2958%20110.517%2064.3902C110.517%2061.4847%20108.161%2059.1292%20105.256%2059.1292C102.35%2059.1292%2099.9946%2061.4847%2099.9946%2064.3902ZM155.632%2064.0889C155.632%2058.3497%20150.804%2054.105%20145.294%2054.105C140.337%2054.105%20135.925%2058.2851%20135.925%2064.3879V80.6993H140.562V71.892C142.107%2073.4081%20143.709%2074.0753%20146%2074.0753C151.177%2074.0753%20155.63%2069.7493%20155.63%2064.0889H155.632ZM151.242%2064.0315C151.242%2066.9371%20148.886%2069.2925%20145.981%2069.2925C143.075%2069.2925%20140.72%2066.9371%20140.72%2064.0315C140.72%2061.126%20143.075%2058.7705%20145.981%2058.7705C148.886%2058.7705%20151.242%2061.126%20151.242%2064.0315Z'%20fill='%2303020D'/%3e%3cpath%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M170.612%2049.2216V53.88C174.831%2054.5211%20178.206%2057.7196%20179.072%2061.8984C179.939%2066.0772%20178.114%2070.3539%20174.497%2072.6196C170.881%2074.8854%20166.236%2074.6618%20162.854%2072.0591L159.069%2075.8447C159.167%2076.1541%20159.22%2076.4764%20159.224%2076.8012C159.223%2078.6337%20157.736%2080.1185%20155.904%2080.118C154.071%2080.1176%20152.586%2078.6322%20152.586%2076.7996C152.585%2074.9671%20154.07%2073.4809%20155.902%2073.4796C156.227%2073.4839%20156.55%2073.5362%20156.859%2073.6351L160.685%2069.8089C158.278%2066.3787%20158.216%2061.8245%20160.53%2058.3303L147.949%2048.5425C147.344%2048.8935%20146.658%2049.0798%20145.959%2049.0829C144.324%2049.081%20142.852%2048.0943%20142.228%2046.5831C141.604%2045.0719%20141.953%2043.3339%20143.11%2042.1796C144.268%2041.0253%20146.007%2040.6822%20147.517%2041.3103C149.026%2041.9383%20150.008%2043.4139%20150.005%2045.0487C150%2045.389%20149.952%2045.7273%20149.862%2046.0555L162.636%2056.0059C164.031%2054.8841%20165.698%2054.1507%20167.467%2053.88V49.2216C166.21%2048.6345%20165.405%2047.3755%20165.398%2045.9885V45.8809C165.398%2043.8998%20167.004%2042.2938%20168.986%2042.2938H169.093C171.074%2042.2938%20172.68%2043.8998%20172.68%2045.8809V45.9885C172.674%2047.3755%20171.868%2048.6345%20170.612%2049.2216ZM163.811%2063.9596C163.809%2066.851%20166.151%2069.1964%20169.043%2069.1991L169.048%2069.1967C171.94%2069.1967%20174.285%2066.852%20174.285%2063.9596C174.283%2061.0682%20171.939%2058.7249%20169.048%2058.7249C166.156%2058.7249%20163.812%2061.0682%20163.811%2063.9596Z'%20fill='%2303020D'/%3e%3c/svg%3e"
           class="mx-auto"
      >
    </li>
    <li class="flex-auto text-center sm:w-1/3 w-1/2">
      <img alt="Webflow"
           src="data:image/svg+xml,%3csvg%20width='244'%20height='122'%20viewBox='0%200%20244%20122'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M96.8205%2057.2614C96.8205%2054.6603%2094.2911%2051.8977%2089.914%2051.8977C84.9988%2051.8977%2079.5992%2055.5034%2078.774%2062.6969C77.9308%2069.9621%2082.4514%2073.1911%2087.0079%2073.1911C91.5644%2073.1911%2093.9503%2071.4152%2096.39%2069.0473C94.2911%2066.4102%2091.5823%2067.6301%2091.0621%2067.8992C90.4881%2068.1862%2089.7526%2068.5808%2088.2457%2068.5808C86.4877%2068.5808%2084.6938%2067.7915%2084.6938%2064.5087C95.9594%2063.3965%2096.8205%2059.8626%2096.8205%2057.2614ZM91.2415%2057.6919C91.1697%2058.4992%2090.8468%2059.8805%2085.1423%2060.6519C86.3442%2056.3824%2088.6404%2056.0595%2089.6988%2056.0595C90.6854%2056.0595%2091.3133%2056.8309%2091.2415%2057.6919ZM72.1545%2058.3916C72.1545%2058.3916%2069.7328%2065.9976%2069.5354%2066.6255C69.4637%2065.9797%2067.6877%2052.3282%2067.6877%2052.3282C63.5618%2052.3282%2061.3553%2055.2702%2060.1893%2058.3736C60.1893%2058.3736%2057.2473%2065.9976%2056.9962%2066.6434C56.9782%2066.0515%2056.5477%2058.4633%2056.5477%2058.4633C56.2965%2054.6603%2052.8164%2052.3462%2050%2052.3462L53.4084%2073.0297C57.7317%2073.0118%2060.0637%2070.0877%2061.2836%2066.9843C61.2836%2066.9843%2063.8847%2060.2572%2063.9923%2059.9522C64.0103%2060.2393%2065.858%2073.0297%2065.858%2073.0297C70.1992%2073.0297%2072.5312%2070.285%2073.787%2067.2892L79.8682%2052.3462C75.5809%2052.3462%2073.3206%2055.2702%2072.1545%2058.3916ZM112.75%2051.8618C110.077%2051.8618%20108.032%2053.3149%20106.31%2055.4496V55.4316L107.853%2043C104.301%2043%20101.413%2046.0855%20100.857%2050.6778L98.1659%2072.8682C100.211%2072.8682%20102.382%2072.2763%20103.548%2070.7694C104.588%2072.1148%20106.149%2073.1911%20108.463%2073.1911C114.454%2073.1911%20118.562%2066.2309%20118.562%2059.7011C118.527%2053.7454%20115.638%2051.8618%20112.75%2051.8618ZM112.194%2062.5354C111.566%2066.177%20109.539%2068.6526%20107.584%2068.6526C105.611%2068.6526%20104.767%2067.7736%20104.767%2067.7736C105.144%2064.5805%20105.377%2062.6251%20106.095%2060.9389C106.812%2059.2526%20108.517%2056.5618%20110.293%2056.5618C112.033%2056.5618%20112.822%2058.8759%20112.194%2062.5354ZM133.523%2052.3462H129.344L129.362%2052.1309C129.649%2049.4042%20130.294%2047.9691%20132.429%2047.7359C133.882%2047.5924%20134.528%2046.8389%20134.689%2046.0137C134.815%2045.3679%20135.21%2043.1256%20135.21%2043.1256C126.814%2043.0718%20124.177%2046.6954%20123.496%2052.2206L123.478%2052.3462H123.388C122.025%2052.3462%20120.518%2053.8889%20120.267%2055.8442L120.177%2056.5618H122.975L120.625%2075.9717L119.908%2079.4518C119.998%2079.4518%20120.105%2079.4698%20120.195%2079.4698C124.123%2079.3262%20126.635%2076.2228%20127.191%2071.7202L129.021%2056.5618H130.33C131.622%2056.5618%20133.129%2055.2702%20133.398%2053.1175L133.523%2052.3462ZM155.265%2051.9874C150.44%2051.9874%20145.883%2055.5393%20144.484%2061.1183C143.085%2066.6973%20145.202%2073.2988%20152.305%2073.2988C159.391%2073.2988%20163.481%2066.4282%20163.481%2060.7057C163.481%2055.0191%20159.625%2051.9874%20155.265%2051.9874ZM157.221%2062.3561C156.97%2064.8854%20155.857%2068.7244%20152.808%2068.7244C149.74%2068.7244%20150.171%2064.2396%20150.476%2062.1049C150.799%2059.8267%20152.072%2056.5797%20154.817%2056.5797C157.275%2056.5797%20157.508%2059.5038%20157.221%2062.3561ZM187.322%2058.3916C187.322%2058.3916%20184.9%2065.9976%20184.703%2066.6255C184.631%2065.9797%20182.855%2052.3282%20182.855%2052.3282C178.729%2052.3282%20176.523%2055.2702%20175.357%2058.3736C175.357%2058.3736%20172.415%2065.9976%20172.164%2066.6434C172.146%2066.0515%20171.715%2058.4633%20171.715%2058.4633C171.464%2054.6603%20167.984%2052.3462%20165.168%2052.3462L168.558%2073.0297C172.881%2073.0118%20175.213%2070.0877%20176.433%2066.9843C176.433%2066.9843%20179.034%2060.2572%20179.124%2059.9522C179.16%2060.2393%20180.99%2073.0297%20180.99%2073.0297C185.331%2073.0297%20187.663%2070.285%20188.919%2067.2892L195%2052.3462C190.731%2052.3462%20188.47%2055.2702%20187.322%2058.3916ZM137.47%2043.1256L133.954%2071.4869L133.236%2074.985C133.326%2074.985%20133.434%2075.003%20133.523%2075.003C137.309%2074.9492%20139.999%2071.6305%20140.502%2067.4507L142.529%2051.1801C143.139%2046.2111%20140.179%2043.1256%20137.47%2043.1256Z'%20fill='%2303020D'/%3e%3c/svg%3e"
           class="mx-auto"
      >
    </li>
    <li class="flex-auto text-center sm:w-1/3 w-1/2">
      <img alt="Mentimeter"
           src="http://127.0.0.1:6006/assets/cust-logo-mentimeter-mono-pos-Bhmht6S_.svg"
           class="mx-auto"
      >
    </li>
    <li class="flex-auto text-center sm:w-1/3 w-1/2">
      <img alt="Toyota"
           src="http://127.0.0.1:6006/assets/cust-logo-toyota-mono-pos-CFVGfjvJ.svg"
           class="mx-auto"
      >
    </li>
    <li class="flex-auto text-center sm:w-1/3 w-1/2">
      <img alt="Split"
           src="data:image/svg+xml,%3csvg%20width='244'%20height='122'%20viewBox='0%200%20244%20122'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M71.2253%2077.7648V87.7101L100.67%2071.0517V61.1064L71.2253%2077.7648Z'%20fill='%2303020D'/%3e%3cpath%20d='M71.2253%2054.1445L100.67%2070.8028V60.8576L71.2253%2044.1992V54.1445Z'%20fill='%2303020D'/%3e%3cpath%20d='M58%2061.8515L87.4448%2078.5099V68.5646L58%2051.9062V61.8515Z'%20fill='%2303020D'/%3e%3cpath%20d='M58%2061.8515L87.4448%2078.5099V68.5646L58%2051.9062V61.8515Z'%20fill='%2303020D'/%3e%3cpath%20d='M87.4448%2035L58%2051.6583V61.6036L87.4448%2044.9453V35Z'%20fill='%2303020D'/%3e%3cpath%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M79.9946%2049.1603L71.2253%2044.1992V54.1213L79.9946%2049.1603ZM78.6755%2073.5494L87.4449%2078.5104V68.5884L78.6755%2073.5494Z'%20fill='%2303020D'/%3e%3cpath%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M158.472%2072.6461H162.694V42.7075H158.472V72.6461ZM186%2054.0766V50.2328H182.305V42.7075H178.082V50.2328H175.037V54.0766H178.082V72.6461H182.305V54.0766H186ZM167.932%2072.6461H172.155V50.2328H167.932V72.6461ZM172.155%2047.6437H167.932V42.7075H172.155V47.6437ZM142.596%2069.3692C147.022%2069.3692%20150.189%2065.769%20150.189%2061.4801C150.189%2057.2318%20146.818%2053.4691%20142.474%2053.4691C138.13%2053.4691%20134.962%2057.192%20134.962%2061.3997C134.962%2065.7284%20138.13%2069.3692%20142.596%2069.3692ZM154.412%2061.7229C154.412%2054.9668%20149.702%2049.6228%20142.759%2049.6228C139.551%2049.6228%20136.628%2050.9986%20134.759%2053.6689H134.676V50.2328H130.821V80.1308H135.044V69.4098H135.127C136.751%2072.0801%20140.039%2073.2122%20143.045%2073.2122C149.496%2073.2122%20154.412%2068.0746%20154.412%2061.7229ZM120.629%2058.7692C124.325%2059.9021%20127.411%2061.7229%20127.411%2066.0516C127.411%2070.4615%20124.162%2073.2122%20119.856%2073.2122C118.873%2073.1984%20117.898%2073.0205%20116.974%2072.6859C113.646%2071.5132%20111.9%2068.8023%20111.9%2065.2825H116.119C116.282%2067.5881%20117.419%2069.3683%20119.936%2069.3683C121.804%2069.3683%20123.184%2068.2354%20123.184%2066.2936C123.184%2064.2537%20121.243%2063.5595%20119.047%2062.7747C116.194%2061.7545%20112.912%2060.5812%20112.912%2056.0997C112.912%2052.2558%20116.282%2049.6261%20119.977%2049.6261C123.836%2049.6261%20126.798%2052.2964%20126.881%2056.1801H122.658C122.455%2054.5623%20121.48%2053.4692%20119.775%2053.4692C118.354%2053.4692%20117.136%2054.3593%20117.136%2055.856C117.136%2057.6957%20118.815%2058.2114%20120.29%2058.6643C120.405%2058.6994%20120.518%2058.7342%20120.629%2058.7692Z'%20fill='%2303020D'/%3e%3c/svg%3e"
           class="mx-auto"
      >
    </li>
    <li class="flex-auto text-center sm:w-1/3 w-1/2">
      <img alt="Australian Open"
           src="http://127.0.0.1:6006/assets/cust-logo-ausopen-mono-pos-aZQqJWfJ.svg"
           class="mx-auto"
      >
    </li>
  </ul>
</section>
`;
